### PR TITLE
1087 adapt parameters_io of ide model

### LINKED ISSUE #1087: Adapt parameters_io of IDE model
We want to make some adaptations on the initialization based on real data in the IDE model regarding the following points:
- RKI reports deaths with respect to the infection date of the respective person. We shift the reported data according to the respective mean stay times to get information on when people died.
- Depending on the chosen transition distributions and the start date we need data points before reporting of Covid cases started. We set these data points to 0. 
- Compute flow ExposedToInfectedNoSymptoms from -2*global_support_max,...,0 to allow correct initialization of flow InfectedNoSymptomsTorecovered.

### PR DESCRIPTION
**Pull request by annawendler**
# Changes and Information

-  RKI reports deaths with respect to the infection date of the respective person. We shift the reported data according to the respective mean stay times to get information on when people died.
- Depending on the chosen transition distributions and the start date we need data points before reporting of Covid cases started. We set these data points to 0.
- Compute flow ExposedToInfectedNoSymptoms from -2*global_support_max,...,0 to allow correct initialization of flow InfectedNoSymptomsTorecovered.

## Merge Request - Guideline Checklist

Please check our [git workflow](https://github.com/SciCompMod/memilio/wiki/git-workflow). Use the **draft** feature if the Pull Request is not yet ready to review.

### Checks by code author

- [x] Every addressed issue is linked (use the "Closes #ISSUE" keyword below)
- [x] New code adheres to [coding guidelines](https://github.com/SciCompMod/memilio/wiki/Coding-guidelines)
- [x] No large data files have been added (files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [x] Tests are added for new functionality and a local test run was successful (with and without OpenMP)
- [x] Appropriate **documentation** for new functionality has been added (Doxygen in the code and Markdown files if necessary)
- [x] Proper attention to licenses, especially no new third-party software with conflicting license has been added
- [ ] (For ABM development) Checked [benchmark results](https://github.com/SciCompMod/memilio/wiki/Agent-Based-Model-Development) and ran and posted a local test above from before and after development to ensure performance is monitored.

### Checks by code reviewer(s)

- [x] Corresponding issue(s) is/are linked and addressed
- [x] Code is clean of development artifacts (no deactivated or commented code lines, no debugging printouts, etc.)
- [x] Appropriate **unit tests** have been added, CI passes, code coverage and performance is acceptable (did not decrease)
- [x] No large data files added in the whole history of commits(files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [x] On merge, add 2-5 lines with the changes (main added features, changed items, or corrected bugs) to the merge-commit-message. This can be taken from the **briefly-list-the-changes** above (best case) or the separate commit messages (worst case).

Closes #1087

### CODE REVIEW

File: cpp/models/ide_secir/parameters_io.cpp [hunk 1]
```diff
@@ -183,10 +232,10 @@ IOResult<void> set_initial_flows(Model& model, ScalarType dt, std::string const&
     }
 
     //--- Calculate the remaining flows. ---
-    // Compute flow ExposedToInfectedNoSymptoms for -global_support_max, ..., 0.
+    // Compute flow ExposedToInfectedNoSymptoms for -2 * global_support_max, ..., 0.
```
Comment (lenaploetzke) [inline cpp/models/ide_secir/parameters_io.cpp:235]: Why is this necessary?

Comment (lenaploetzke) [reply cpp/models/ide_secir/parameters_io.cpp:235]: Ah to calculate the InfectedNoSymptomsToRecovered transition, right?

Comment (annawendler) [reply cpp/models/ide_secir/parameters_io.cpp:235]: Yes

File: cpp/tests/test_ide_parameters_io.cpp [hunk 2]
```diff
@@ -117,6 +121,24 @@ TEST(TestIDEParametersIo, ParametersIoRKIFailure)
...
+    start_date = mio::Date(2020, 5, 24);
+    status =
+        mio::isecir::set_initial_flows(model, dt, mio::path_join(TEST_DATA_DIR, "cases_all_germany.json"), start_date);
+    // Check that status is Success as just a warning is logged.
+    ASSERT_THAT(print_wrap(status), IsSuccess());
+    // Check that the affected flows have actually been set to 0.
+    // As start_date is the first date in the data file, there is some infection data for all times >-1
+    // (as we interpolate linearly in between -1 and 0). Therefore, we only expect the flows to be zero for times <=-1.
+    for (Eigen::Index i = 0; i < model.m_transitions.get_num_time_points() - 2; i++) {
+        // Check only the flows "after" InfectedNoSymptomsToInfectedSymptoms where earlier data is needed.
```
Comment (annawendler) [inline cpp/tests/test_ide_parameters_io.cpp:134]: Why not from InfectedNoSymptomsToInfectedSymptoms on?

**Review by lenaploetzke — APPROVED** [review]

**Review by lenaploetzke — APPROVED** [review]
Thank you for adapting the number of deaths and especially for finding the bug!
